File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) 
    {
        Set<Character> num = new HashSet<>();
        int maxLen = 0, left = 0;

        for(int i=0; i<s.length(); i++)
        {
            while(num.contains(s.charAt(i)))
            {
                num.remove(s.charAt(left));
                left++;
            }
                maxLen = Math.max(maxLen, i-left+1);
                num.add(s.charAt(i));
        }    
        return maxLen;
    }
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int left = 0;
        int ans = 0;
        unordered_map<char, int> map;

        for(int i=0; i<s.size(); i++){
            char ch = s[i];
            // character in map
            while (map.count(ch) != 0) {
            map.erase(s[left]); 
            left++;
            }

            // not in map
            map[s[i]] = 1;
            ans = max(ans, i-left+1);
        }
        return ans;
    }
};